File: ResourceManager/Models/Role.vb
Imports ResourceManager.Models.Abstractions
Imports System.Collections.Generic

Namespace ResourceManager.Models
    Public Class Role
        Inherits BaseEntity
        Implements IItem
        Public Property Name As String Implements IItem.Name
        Public Property Description As String
        Public Overridable Property UserRoles As ICollection(Of UserRole)
        Public Overridable Property RolePermissions As ICollection(Of RolePermission)
        Public Overrides Function GetCaption() As String
            Return String.Format("{0} with Name '{1}'", NameOf(Role), Name)
        End Function

    End Class
End Namespace

File: ResourceManager/Models/Role.cs
using ResourceManager.Models.Abstractions;
using System;
using System.Collections.Generic;

namespace ResourceManager.Models
{
    public class Role : BaseEntity, IItem
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public virtual IList<UserRole> UserRoles { get; set; } = new List<UserRole>(); 
        public virtual IList<RolePermission> RolePermissions { get; set; } = new List<RolePermission>();
        public override string GetCaption()
        {
            return string.Format("{0} with Name '{1}'", nameof(Role), Name);
        }

    }
}
